### PR TITLE
[airlearner] first commit for airlearner

### PR DESCRIPTION
**Pull request by jq**
@jieyingc

once this merge, we will add more docs for how to use it

### CODE REVIEW

**Comment by jieyingchen** [comment]
LGTM

**Comment by mLewisLogic** [comment]
```
➜  airlearner git:(master) gradlew test
:compileJava NO-SOURCE
:processResources NO-SOURCE
:classes UP-TO-DATE
:compileTestJava NO-SOURCE
:processTestResources NO-SOURCE
:testClasses UP-TO-DATE
:test NO-SOURCE
:utils:compileJava
/Users/mike_lewis/code/xgboost/jvm-packages/contributions/airlearner/utils/src/main/java/com/airbnb/common/config/AirCon.java:22: error: cannot find symbol
  protected MacroExecutor executor = null;
            ^
  symbol:   class MacroExecutor
  location: class AirCon
/Users/mike_lewis/code/xgboost/jvm-packages/contributions/airlearner/utils/src/main/java/com/airbnb/common/config/AirCon.java:52: error: cannot find symbol
  protected AirCon(MacroExecutor executor) {
                   ^
  symbol:   class MacroExecutor
  location: class AirCon
2 errors
:utils:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':utils:compileJava'.
```

**Comment by jq** [comment]
@mLewisLogic  I am able to run ` gradlew test ` without a problem. MacroExecutor is under the same folder.

File: jvm-packages/contributions/airlearner/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bintray_user=tingtone1
+bintray_key=04728afec7da04164bb11fae1607f5b458d89251
```
Comment (mLewisLogic) [inline jvm-packages/contributions/airlearner/gradle.properties]: @jq We need to rotate and protect this key immediately. Anybody can publish to bintray using this key. It likely hasn't caused any issues yet, but it's an open security hole given that we run this code within our production cluster.

File: jvm-packages/contributions/airlearner/strategy/build.gradle [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+apply from: "$rootDir/configs/version.gradle"
+apply plugin: 'com.jfrog.bintray'
+
+repositories {
+  mavenLocal()
+}
+
+dependencies {
+  compile 'com.typesafe:config:1.3.0'
+  compile project(':utils')
```
Comment (mLewisLogic) [inline jvm-packages/contributions/airlearner/strategy/build.gradle]: How does this work? Specifically, compiling in the HEAD utils code, when `utils` itself is versioned and being included in other dependencies by its version number.

File: jvm-packages/contributions/airlearner/gradle.properties (hunk 1 above)
Comment (mLewisLogic) [inline jvm-packages/contributions/airlearner/gradle.properties]: Hopefully this is a repo-specific key and isn't used for any other Airbnb projects?

**Pull request merged**